pymeasure/display/inputs.py

```diff
@@ -55,8 +55,8 @@ def set_parameter(self, parameter):
         """
         self._parameter = parameter
 
-        if parameter.default is not None:
-            self.setValue(parameter.default)
+        if parameter.value is not None:
+            self.setValue(parameter.value)
```

Comment:
Hello Casper,

first of all, I agree with You that this is a workaround to please the test more than a proper solution.
I did some research on python unittest module and I have tried to figure out the rationale behind the test.
The test is trying to assess that when you use the BooleanInput widget and change its value, the property 'value' associated is updated exactly once and with the expected value.
Since during the class instantiation ("BooleanInput(bool_param)"), the 'value' property is read twice, this is accounted as two calls with no parameters in the mock.
My proposal is to add the following code to fix the test (see line "p.reset_mock()"):

    def test_leftclick_should_update_parameter(self, qtbot, default_value):
        # set up BooleanInput

        bool_param = BooleanParameter(potato, default=default_value)
        with mock.patch('test_inputs.BooleanParameter.value',
                new_callable=mock.PropertyMock,
                return_value=default_value) as p:
            bool_input = BooleanInput(bool_param)

            # Clear any call to property 'value' during initialization
            p.reset_mock()

            qtbot.addWidget(bool_input)
            bool_input.show()

            # TODO: fix: fails to toggle on Windows
            #qtbot.mouseClick(bool_input, QtCore.Qt.LeftButton)
            bool_input.setValue(not default_value)

            assert bool_input.value() == (not default_value)
            bool_input.parameter # lazy update
            p.assert_called_once_with(not default_value)

Does this make sense to You ?

Michele

Reply:
I agree that we should not change the implementation to "please the test".
As to why the tests asserts on one call, maybe this was just an oversight in the test design shadowed by the bug you fix in here. @Laogeodritt implemented this test, maybe you can comment?
The proposed change to the test sounds reasonable to me. Great that you already thought of a comment explaining why that line is in there.

Reply:
Thanks for diving in Michele! I agree that the proposed fix in the test sounds reasonable and clear.